Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -2696,7 +2696,7 @@ WebIDL::ExceptionOr<String> Document::cookie(Cookie::Source source)
 
     // Otherwise, the user agent must return the cookie-string for the document's URL for a "non-HTTP" API, decoded using
     // UTF-8 decode without BOM.
-    return page().client().page_did_request_cookie(m_url, source);
+    return Infra::isomorphic_decode(page().client().page_did_request_cookie(m_url, source));
```

Comment:
Hmm, the spec explicitly mentions doing UTF8 decode here (i.e: String::from_utf8_with_replacement_character)

Reply:
Yeah that's very odd, the step links to a part of the [HTTP spec](https://httpwg.org/specs/rfc6265.html#sane-cookie-syntax) which defines `cookie-string`s to consist of a subset ASCII.